lnrpc/walletrpc/walletkit_server_test.go

```diff
@@ -0,0 +1,59 @@
+//go:build walletrpc
+// +build walletrpc
+
+package walletrpc
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/stretchr/testify/require"
+)
+
+// TestWitnessTypeMappings tests that the mapping between the witness type
+// enum defined in the `input` package and the witness type enum protocol
+// buffer definition in the `walletrpc` package is correct.
+func TestWitnessTypeMapping(t *testing.T) {
+	t.Parallel()
+
+	require.Equal(
+		t,
+		len(allWitnessTypes),
+		len(WitnessType_name)-1, // don't count default value
+		"number of witness types should match proto definition",
+	)
+
+	// test that for each individual witness type mapping 1) the mapped
+	// witness types have the same numerical value, and 2) the string
+	// representations are equivalent.
+	for witnessType, witnessTypeProto := range allWitnessTypes {
+		witnessType, ok := witnessType.(input.StandardWitnessType)
```

Comment:
Not sure about this type check, why restrict the allWitnessTypes to StandardWitnessType, then we could also just use StandardWitnessType it the `allWitnessTypes` in the first place.

Reply:
agreed, there is no need to restrict `allWitnessTypes` to `StandardWitnessType` only.

I cast them to `StandardWitnessType` in order to compare the numbering. I've changed the test so that the numbering comparison only happens if `witnessType` *is* a `StandardWitnessType`.